File: app/src/main/java/ro/ubb/bigbucks/repository/ExpenseRepository.kt
package ro.ubb.bigbucks.repository

import ro.ubb.bigbucks.model.Expense

interface ExpenseRepository {
    fun add(expense: Expense): Expense

    fun getById(id: UInt): Expense?

    fun getAll(): List<Expense>

    fun size(): Int

    fun deleteById(id: UInt): Boolean

    fun deleteAll()
}

File: big_bucks_app_non_native/lib/repository/abstract/expense_repository.dart
import 'package:big_bucks_app/model/expense.dart';

class ExpenseError implements Exception {
  final String message;

  const ExpenseError(this.message);

  @override
  String toString() {
    return 'ExpenseError: $message';
  }
}

abstract class ExpenseRepository {
  const ExpenseRepository();

  Future<Expense> add(Expense expense);

  Future<Expense?> getById(int expenseId);

  Future<List<Expense>> getAll();

  Future<bool> update(Expense expense);

  Future<bool> deleteById(int expenseId);

  Future<void> deleteAll();
}
